### PR TITLE
[integration] added nigthly action (for creating tarball of integration and master)

### PR DESCRIPTION
**Pull request by fstagni**
This PR adds a GitHub Actions workflow that builds and uploads, every 24 hours, an "integration" and a "master" tarball. The "integration" tarball is then used for checks on PRs targeting the integration branch.

### CODE REVIEW

**Comment by fstagni** [comment]
Review, please.

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  dirac-distribution-integration:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: login
+        run: echo ${{ secrets.GITHUB_TOKEN }} | docker login docker.pkg.github.com --username ${{ github.actor }} --password-stdin
+      - name: create
+        run: |
+          docker pull docker.pkg.github.com/diracgrid/management/dirac-distribution:latest
+          docker run docker.pkg.github.com/diracgrid/management/dirac-distribution:latest bash -c "python3 dirac-distribution.py -r integration  | tail -n 1  > /tmp/deploy.sh && sed -i 's/lhcbprod/${{ secrets.KRB_USERNAME }}/g' /tmp/deploy.sh && cat /tmp/deploy.sh && echo ${{ secrets.KRB_PASSWORD }} | kinit ${{ secrets.KRB_USERNAME }}@CERN.CH && echo readyToUpload && export USER=${{ secrets.KRB_USERNAME }} &&  echo reallyReadyToUpload && source /tmp/deploy.sh"
```
Comment (chaen) [inline .github/workflows/nightly.yml]: Just to make it more readable, can it go on several lines ? :-D

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# This workflow runs every 24 hours for the following purpose:
+#
+# - Create + upload *integration* tarball
+# 
+# All created images are then uploaded to GitHub packages
+
+name: Nighlies
```
Comment (andresailer) [inline .github/workflows/nightly.yml]:
```suggestion
name: Nightlies
```

**Comment by andresailer** [comment]
Does this mean that each forked repo with enabled actions would also run a cron job, that would probably fail, every night?

**Comment by fstagni** [comment]
Good question @andresailer you are not the only one asking: https://github.community/t5/GitHub-Actions/Execute-master-branch-and-specific-Git-tag-on-schedule/m-p/41715/highlight/true#M4724 but without an answer. I have asked this specific question in the same forum.

**Comment by fstagni** [comment]
I might have found a simple solution... (see last commit)

**Comment by andresailer** [comment]
> I might have found a simple solution... (see last commit)

While this prevents the job from running, it would still give an error because now there is an action, but no job. Similar to what happens on forked repositories at the moment because of

https://github.com/DIRACGrid/DIRAC/blob/2447450ec3e421821f6dead4cd729c77e3bd6277/.github/workflows/integration.yml#L8

e.g.: https://github.com/fstagni/DIRAC/commit/ac8a7e9a9b1a3ea1496419e17d0e4a9ed1bfca77/checks?check_suite_id=391829328

It seems that the only solution to avoid failing actions is to add an dummy job:
https://github.community/t5/GitHub-Actions/Workflow-is-failing-if-no-job-can-be-ran-due-to-condition/m-p/38186#M3250

For my fork of DIRAC  I have now disabled Actions.

**Comment by fstagni** [comment]
I know. Basically, you just receive a mail that your workflow is failing. I don't want to add a dummy job just because GitHub does not treat properly this case. Let's hope it will, one day.

**Comment by fstagni** [comment]
So, I can confirm that there's no way to restrict per fork: https://github.community/t5/GitHub-Actions/Restrict-execution-of-workflows-on-a-specific-organization/td-p/42832 so what I coded is the only "solution" that we can have.

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    if: github.repository == 'DIRACGrid/DIRAC'
+    steps:
+      - uses: actions/checkout@v1
+      - name: login
+        run: echo ${{ secrets.GITHUB_TOKEN }} | docker login docker.pkg.github.com --username ${{ github.actor }} --password-stdin
+      - name: create
+        run: |
+          docker pull docker.pkg.github.com/diracgrid/management/dirac-distribution:latest
+          docker run docker.pkg.github.com/diracgrid/management/dirac-distribution:latest bash -c "python3 dirac-distribution.py -r integration  | tail -n 1  > /tmp/deploy.sh && sed -i 's/lhcbprod/${{ secrets.KRB_USERNAME }}/g' /tmp/deploy.sh && cat /tmp/deploy.sh && echo ${{ secrets.KRB_PASSWORD }} | kinit ${{ secrets.KRB_USERNAME }}@CERN.CH && echo readyToUpload && export USER=${{ secrets.KRB_USERNAME }} &&  echo reallyReadyToUpload && source /tmp/deploy.sh"
```
Comment (chrisburr) [inline .github/workflows/nightly.yml]: How about:
```suggestion
          docker run docker.pkg.github.com/diracgrid/management/dirac-distribution:latest bash -c \
          "python3 dirac-distribution.py -r integration | tail -n 1  > /tmp/deploy.sh && "\
          "sed -i 's/lhcbprod/${{ secrets.KRB_USERNAME }}/g' /tmp/deploy.sh && "\
          "cat /tmp/deploy.sh && "\
          "echo ${{ secrets.KRB_PASSWORD }} | kinit ${{ secrets.KRB_USERNAME }}@CERN.CH && "\
          "echo readyToUpload && "\
          "export USER=${{ secrets.KRB_USERNAME }} && "\
          "echo reallyReadyToUpload && "\
          "source /tmp/deploy.sh"
```

**Comment by fstagni** [comment]
Review OK?

**Comment by fstagni** [comment]
I would need to have this merged. Are there objections?

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+jobs:
+  # running dirac-distribution in the proper image
+  dirac-distribute:
+    runs-on: ubuntu-latest
+    if: github.repository == 'DIRACGrid/DIRAC'
+
+    strategy:
+      fail-fast: False
+      matrix:
+	branch:
```
Comment (chrisburr) [inline .github/workflows/nightly.yml]: The whitespace in this file is a mix of tabs and spaces

**Comment by atsareg** [comment]
I can live the pain of getting mails nightly from my fork. So, it is OK with me

**Comment by fstagni** [comment]
I am merging now.

**Pull request merged**